pkg/operator/cassandra/controller/controller.go

```diff
@@ -0,0 +1,304 @@
+	recorder record.EventRecorder
+}
+
+// New returns a new ClusterController
+func New(
+	rookImage string,
+	kubeClient kubernetes.Interface,
+	rookClient rookClientset.Interface,
+	clusterInformer informersv1alpha1.ClusterInformer,
+	statefulSetInformer appsinformers.StatefulSetInformer,
+	serviceInformer coreinformers.ServiceInformer,
+	podInformer coreinformers.PodInformer,
+	serviceAccountInformer coreinformers.ServiceAccountInformer,
+	roleInformer rbacinformers.RoleInformer,
+	roleBindingInformer rbacinformers.RoleBindingInformer,
+
+) *ClusterController {
+
+	// Add sample-controller types to the default Kubernetes Scheme so Events can be
+	// logged for sample-controller types.
+	rookScheme.AddToScheme(scheme.Scheme)
+	// Create event broadcaster
+	logger.Infof("creating event broadcaster...")
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(logger.Infof)
+	eventBroadcaster.StartRecordingToSink(&typedcorev1.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")})
+	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: controllerName})
+
+	cc := &ClusterController{
+		rookImage:  rookImage,
+		kubeClient: kubeClient,
+		rookClient: rookClient,
+
+		clusterLister:              clusterInformer.Lister(),
+		clusterListerSynced:        clusterInformer.Informer().HasSynced,
+		statefulSetLister:          statefulSetInformer.Lister(),
+		statefulSetListerSynced:    statefulSetInformer.Informer().HasSynced,
+		podLister:                  podInformer.Lister(),
+		podListerSynced:            podInformer.Informer().HasSynced,
+		serviceLister:              serviceInformer.Lister(),
+		serviceListerSynced:        serviceInformer.Informer().HasSynced,
+		serviceAccountLister:       serviceAccountInformer.Lister(),
+		serviceAccountListerSynced: serviceAccountInformer.Informer().HasSynced,
+		roleLister:                 roleInformer.Lister(),
+		roleListerSynced:           roleInformer.Informer().HasSynced,
+		roleBindingLister:          roleBindingInformer.Lister(),
+		roleBindingListerSynced:    roleBindingInformer.Informer().HasSynced,
+
+		queue:    workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), clusterQueueName),
+		recorder: recorder,
+	}
+
+	// Add event handling functions
+
+	clusterInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			newCluster := obj.(*cassandrav1alpha1.Cluster)
+			cc.enqueueCluster(newCluster)
+		},
+		UpdateFunc: func(old, new interface{}) {
+			newCluster := new.(*cassandrav1alpha1.Cluster)
+			oldCluster := old.(*cassandrav1alpha1.Cluster)
+			if newCluster.ResourceVersion == oldCluster.ResourceVersion {
+				return
+			}
+			cc.enqueueCluster(newCluster)
+		},
+		DeleteFunc: func(obj interface{}) {
+			// TODO: handle deletion
+		},
+	})
+
+	statefulSetInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
that's interesting that we are watching statefulset changes too.  is that scoped to just the ones we expect this operator to be in charge of or is it all statefulsets? i'm curious to read further and see what this controller does with the stateful set events